package/homebrew.spec

```diff
@@ -0,0 +1,90 @@
+%define homebrew_user linuxbrew
+%define homebrew_directory /home/%{homebrew_user}/.%{homebrew_user}
+%define brew_repo https://github.com/Homebrew/brew
+
+Summary:   Homebrew package manager
+Name:      homebrew
+Version:   0.0.0
+Release:   %autorelease
+License:   BSD-2-clause
+Group:     Development/Tools
+URL:       https://brew.sh
+Source:    %{brew_repo}/archive/refs/tags/%{git_version}.tar.gz#/brew.tar.gz
+
+# See: https://github.com/Homebrew/install/blob/master/install.sh#L211-L214
+BuildRequires: git
+
+Requires: procps-ng
+Requires: file
+Requires: gcc
+Requires: git >= 2.7.0
+Requires: glibc >= 2.13
+Requires: ruby >= 2.6.0
+
+%description
+The Missing Package Manager for macOS (or Linux)
+
+%global debug_package %{nil}
+%global _missing_build_ids_terminate_build 0
+%global __brp_mangle_shebangs /usr/bin/true
+
+%prep
+%autosetup -n brew-%{version}
+
+%build
+git init
+git remote set-url origin %{brew_repo} || git remote add origin %{brew_repo}
+
+%install
+install -d "$RPM_BUILD_ROOT%{homebrew_directory}"
+cp -r bin "$RPM_BUILD_ROOT%{homebrew_directory}"
+cp -r completions "$RPM_BUILD_ROOT%{homebrew_directory}"
+cp -r manpages "$RPM_BUILD_ROOT%{homebrew_directory}"
+cp -r docs "$RPM_BUILD_ROOT%{homebrew_directory}"
+cp -r package "$RPM_BUILD_ROOT%{homebrew_directory}"
+cp -r .git "$RPM_BUILD_ROOT%{homebrew_directory}"
```

Comment:
Packaging and maintaining the `.git` folder might be problematic since there will then some clash when the package is updated. I am asking the Fedora people for some advice about this.

Reply:
One suggestion from @Conan-Kudo on the matrix chat is to disable the updating process from the `brew` command, i.e. have all of the changes be owned by the rpm spec. Would that be possible or desired upstream?

Reply:
Ok, will have to test how the `%ghost` or other directives should be used. I think there are examples like `nextcloud` that do updates independent of the Fedora package, but that one does not publish the `.git`. What about having `brew` (or rather a small wrapper of `/usr/bin/brew`) run the `git init` when needed like Neal suggested.

BTW: does `brew` work when it is run from `/usr/bin/brew` instead of `/home/linuxbrew/bin/brew`?

Reply:
I think we need to avoid modifications of normal invocations of `brew` itself. `brew update` already handles this sort of initialisation and it's run automatically so that's the best thing to test with.

> BTW: does `brew` work when it is run from `/usr/bin/brew` instead of `/home/linuxbrew/bin/brew`?

If you just move files or symlink them: no. You could create a wrapper script but it's probably going to have some permission issues. Better for folks to just add the `linuxbrew` directory into their `PATH`.

Reply:
Probably not, it needs to be `/home/linuxbrew/.linuxbrew/bin/brew` even

Reply:
> If you just move files or symlink them: no. You could create a wrapper script but it's probably going to have some permission issues. Better for folks to just add the linuxbrew directory into their PATH.

This is a bit tricky, but I guess some other packages like `intel` compiler use non-OS standards as well in the spec file. At the review stage will need to discuss if it can make it into `Fedora`/`RHEL` downstream in that configuration. Maybe with having `/usr/bin/brew` as a thin wrapper to `sudo -u linuxbrew -g linuxbrew /home/linuxbrew/bin/brew` it can work, if we provide a reasonable `/etc/sudoer` file.

> I think we need to avoid modifications of normal invocations of brew itself. brew update already handles this sort of initialisation and it's run automatically so that's the best thing to test with.

The issue I am concerned here is when running `brew update` and at some other point `dnf upgrade homebrew`, the latter will overwrite the files from the former if the package is owning those files. One way around that is to not have the package own the files that could be changed by that (maybe even no files at all other than the wrapper). That could be done by setting the installed files there as `%ghost` iiuc (that would allow `dnf remove homebrew` to remove the home directory). Will need to run rpminspect and other automation tools to confirm there are no unusual behaviours due to this

Reply:
> Maybe with having `/usr/bin/brew` as a thin wrapper to `sudo -u linuxbrew -g linuxbrew /home/linuxbrew/bin/brew` it can work, if we provide a reasonable `/etc/sudoer` file.

It's much more involved than this. Not every command works reliably like this (I've tried). Homebrew is simply not designed to work in multi-user mode where it's not owned by a single user.

> The issue I am concerned here is when running `brew update` and at some other point `dnf upgrade homebrew`, the latter will overwrite the files from the former if the package is owning those files.

Yup, that seems to be reasonable behaviour.

Reply:
For some of them, yes. For every part of the brew tool this would be fine.